integration/test/dashboard_test.go

```diff
@@ -0,0 +1,130 @@
+//+build regression
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/Netapp/harvest-automation/test/docker"
+	"github.com/Netapp/harvest-automation/test/utils"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	"log"
+	"testing"
+)
+
+const TOTAL_DASHBOARD = 11
+
+type DashboardImportTestSuite struct {
+	suite.Suite
+}
+
+type Folder struct {
+	Id    int64  `json:"id"`
+	Uid   string `json:"uid"`
+	Title string `json:"title"`
+}
+
+type Dashboard struct {
+	Id          int64  `json:"id"`
+	Uid         string `json:"uid"`
+	Title       string `json:"title"`
+	FolderTitle string `json:"folderTitle"`
+	FolderUrl   string `json:"folderUrl"`
+}
+
+func (suite *DashboardImportTestSuite) SetupSuite() {
+	log.Println("Verify Grafana and Prometheus are configured")
+	if !utils.IsUrlReachable(utils.GetGrafanaHttpUrl()) {
+		panic(fmt.Errorf("Grafana is not reachable."))
+	}
+	if !utils.IsUrlReachable(utils.GetPrometheusUrl()) {
+		panic(fmt.Errorf("Grafana is not reachable."))
+	}
+	log.Println("Import dashboard from grafana/dashboards")
+	params := []string{"import", "--addr", utils.GetGrafanaUrl(), "--directory", "grafana/dashboards"}
+	containerID := docker.GetContainerID("poller")
+	if len(containerID) == 0 {
+		//assuming non docker based harvest
+		log.Println("It is non docker based harvest")
+		importOutput := utils.Exec("/opt/harvest", "bin/grafana", params...)
+		log.Println(importOutput)
+	} else {
+		execParam := []string{"exec", containerID, "bin/grafana"}
+		for index, _ := range params {
+			execParam = append(execParam, params[index])
+		}
+		importOutput := utils.Run("docker", execParam...)
+		log.Println(importOutput)
+	}
+
+}
+
+func (suite *DashboardImportTestSuite) TestImport() {
```

Comment:
Would be nice to relate this test to the issue in the repo that caused us to add this test

Reply:
added